tests/test_salesforce_bookmarks.py

```diff
@@ -1,165 +1,41 @@
-
-                # Verify the first sync sets a bookmark of the expected form
-                self.assertIsNotNone(first_bookmark_key_value)
-                self.assertIsNotNone(first_bookmark_key_value.get(replication_key))
-
-                # Verify the second sync sets a bookmark of the expected form
-                self.assertIsNotNone(second_bookmark_key_value)
-                self.assertIsNotNone(second_bookmark_key_value.get(replication_key))
-
-                # bookmarked states (actual values)
-                first_bookmark_value = first_bookmark_key_value.get(replication_key)
-                second_bookmark_value = second_bookmark_key_value.get(replication_key)
-                # bookmarked values as utc for comparing against records
-                first_bookmark_value_utc = self.convert_state_to_utc(first_bookmark_value)
-                second_bookmark_value_utc = self.convert_state_to_utc(second_bookmark_value)
-
-                # Verify the second sync bookmark is Equal to the first sync bookmark
-                self.assertEqual(second_bookmark_value, first_bookmark_value) # assumes no changes to data during test
-
-                # Verify the second sync records respect the previous (simulated) bookmark value
-                simulated_bookmark_value = new_states['bookmarks'][stream][replication_key]
-                for record in second_sync_messages:
-                    replication_key_value = record.get(replication_key)
-                    self.assertGreaterEqual(replication_key_value, simulated_bookmark_value,
-                                            msg="Second sync records do not repect the previous bookmark.")
-
-                # Verify the first sync bookmark value is the max replication key value for a given stream
-                for record in first_sync_messages:
-                    replication_key_value = record.get(replication_key)
-                    self.assertLessEqual(replication_key_value, first_bookmark_value_utc,
-                                         msg="First sync bookmark was set incorrectly, a record with a greater rep key value was synced")
-
-                # Verify the second sync bookmark value is the max replication key value for a given stream
-                for record in second_sync_messages:
-                    replication_key_value = record.get(replication_key)
-                    self.assertLessEqual(replication_key_value, second_bookmark_value_utc,
-                                         msg="Second sync bookmark was set incorrectly, a record with a greater rep key value was synced")
-
-                # Verify the number of records in the 2nd sync is less then the first
-                self.assertLess(second_sync_count, first_sync_count)
+    bookmark_format ="%Y-%m-%dT%H:%M:%S.%fZ"
+
+    initial_bookmarks = {}
+    streams_replication_method = {}
+    def streams_replication_methods(self):
```

Comment:
Wondering if these two methods are specific to the bookmarks test or should be moved to sfbase

Reply:
For now I think its better to leave them in bookmark test, I don't see other test which requires these methods for now.